docs/guidelines/git-github.md

```diff
@@ -0,0 +1,306 @@
+
+
+!!! tip "Use SSH with Git"
+
+    We recommend using SSH when Git authenticates and communicates with github.com. Private keys should be passphrase protected
+
+### Configuring Git to use SSH
+
+The [Connecting to GitHub with SSH](https://docs.github.com/en/authentication/connecting-to-github-with-ssh) in the official Github documentation is a good source for detailed information.
+
+The following sections of the guideline contains the usual steps to get started with SSH.
+
+#### [Generate a new SSH key](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#generating-a-new-ssh-key)
+  - Type of key should be `ed25519`
+  - Specify a filename (`-f`). We assume you can have multiple keys and suggest a naming convention like "service"-"identity"-"index". The index could be creation date and then used for automating key rotation.
+  - Add a comment (`-C`) to help identify the key's purpose. This will be added to the public key as generic text but will not have any functional impact. It will not be exposed on github.com. However, the purpose of a public key is to "share it", so use with caution (PII).
+
+Example; Creating a SSH key for the Github user `larskaare`
+
+```shell
+ssh-keygen -t ed25519 -f ~/.ssh/github_larskaare_1 -C "Github SSH auth key for machine 1"
+```
+
+Two files are created, one named `github_larskaare_1` and one named `github_larskaare_1.pub`. The file with the `.pub` extension contains the public part of the key.
+
+!!! Tip "Re-using keys?"
+
+    We advice on creating separate SSH keys for separate machines and devices and not to re-use the same key on them all.
+
+!!! Tip "Passphrases"
+
+    Store passphrases in a password manager.
+
+#### Configure SSH and adding the key to the key-agent
+
+Adding the generated SSH key to the `ssh-agent` gives you a secure repository for your private keys's passphrases. Adding keys and passphrases to the key agent eliminates the need to repeatedly enter the passphrase.
+
+Follow the [official documentation](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#adding-your-ssh-key-to-the-ssh-agent) of and add the SSH key to the ssh-agent. Be aware of the operating system selector at the top of the page - it will give you instructions for **Mac**, **Windows** and **Linux**.
+
+#### Adding SSH config
+
+SSH uses a config file for it's configuration (this file is not used by Git). The user-specific file is usually stored in `~/.ssh/config` and should be updated prior to using the SSH key and the ssh-key agent. Consult the documentation of your SSH client for details. (Many use OpenSSH which have good manual pages - [OpenSSH Manual Pages](https://www.openssh.com/manual.html))
+
```

Comment:
Should this part be moved up?
Above one is using `ssh-key` and then here one states that this part should be performed _before_....

Reply:
Will leave as-is for now. It's a good point in the "guide" to include the ssh config file as it follows a natural progression of a "tutorial" - imho :)